mailpoet/lib/Segments/DynamicSegments/FilterFactory.php

```diff
@@ -145,15 +153,25 @@ private function email($action) {
     return $this->emailAction;
   }
 
+  /**
+   * @return WooCommerceMembership
+   */
   private function wooCommerceMembership() {
     return $this->wooCommerceMembership;
   }
 
+  /**
+   * @return WooCommerceSubscription
+   */
   private function wooCommerceSubscription() {
     return $this->wooCommerceSubscription;
   }
 
-  private function wooCommerce($action) {
+  /**
+   * @param ?string $action
+   * @return WooCommerceCategory|WooCommerceCountry|WooCommerceNumberOfOrders|WooCommerceProduct|WooCommerceTotalSpent
+   */
+  private function wooCommerce(?string $action) {
```

Comment:
I wonder if it is better to be specific like `WooCommerceCategory|WooCommerceCountry|WooCommerceNumberOfOrders|WooCommerceProduct|WooCommerceTotalSpent` or just use the interface `Filter` as return type. What do you think?

Reply:
Agreed and done.